roles/kubernetes-apps/external_cloud_controller/openstack/tasks/openstack-credential-check.yml

```diff
@@ -4,24 +4,63 @@
     msg: "external_openstack_auth_url is missing"
   when: external_openstack_auth_url is not defined or not external_openstack_auth_url
 
-- name: External OpenStack Cloud Controller | check external_openstack_username value
+
+- name: External OpenStack Cloud Controller | check external_openstack_username or external_openstack_application_credential_name value
+  fail:
+    msg: "you must either set external_openstack_username or external_openstack_application_credential_name"
+  when:
+    - external_openstack_username is not defined or not external_openstack_username
```

Comment:
Why check that external_openstack_username is not boolean 'False'?
Since the default is for external_openstack_username to be undefined, isn't it sufficient to make sure it is defined?
Likewise for external_openstack_application_credential_name.

Reply:
`external_openstack_username is not defined or not external_openstack_username` is taken from existing code. The idea was to add more tests related to application credentials instead of refactoring all the tests. But you're right, those conditions could be improved as the role's default would in any case define `external_openstack_username` variable.

`external_openstack_application_credential_name`doesn't have a default because it is not very likely to be defined in a user's environment (like `OS_USERNAME` could be).

And beware that the `bool` filter can only be applied to defined variables. Ansible would throw an error otherwise.

The condition needs to make sure that either `external_openstack_username` or `external_openstack_application_credential_name` is defined and is not empty.